cli/util/upload_state.go

```diff
@@ -193,14 +193,14 @@ func searchStateIndex(ctx *cli.Context, p neofs.PoolWrapper, containerID cid.ID,
 }
 
 func traverseMPT(root util.Uint256, store storage.Store, writer *gio.BinWriter) error {
-	cache := storage.NewMemCachedStore(store)
-	billet := mpt.NewBillet(root, mpt.ModeAll, mpt.DummySTTempStoragePrefix, cache)
-	err := billet.Traverse(func(pathToNode []byte, node mpt.Node, nodeBytes []byte) bool {
-		writer.WriteVarBytes(nodeBytes)
-		return writer.Err != nil
-	}, false)
-	if err != nil {
-		return fmt.Errorf("billet traversal error: %w", err)
-	}
-	return nil
+	trieStore := mpt.NewTrieStore(root, mpt.ModeAll, store)
+	prefix := []byte{byte(storage.STStorage)}
+	rng := storage.SeekRange{Prefix: prefix}
+
+	trieStore.Seek(rng, func(k, v []byte) bool {
+		writer.WriteVarBytes(k)
```

Comment:
You need to strip the storage prefix itself, it's an implementation-level thing. This comment is relevant only if you don't use stateroot module.